autrainer/core/utils/bookkeeping.py

```diff
@@ -101,6 +101,76 @@ def create_folder(self, folder_name: str, path: str = "") -> None:
             exist_ok=True,
         )
 
+    def save_dataset_summary(
+        self,
+        data: "AbstractDataset",
+        filename: str,
+        val_exists: bool = False,
+    ) -> None:
+        """Save some overview of the dataset to a file.
+
+        Args:
+            data: dataset object
+            filename: Name of the file to save the overview to.
+        """
+        if val_exists:
+            split_datasets = [data.train_dataset, data.val_dataset, data.test_dataset]
+            index = ["Train", "Development", "Test", "Total"]
+        else:
+            split_datasets = [data.train_dataset, data.test_dataset]
+            index = ["Train", "Test", "Total"]
+        n_samples = []
+        if isinstance(data, BaseClassificationDataset) or isinstance(data, BaseMLClassificationDataset):
+            n_classes = []
+            n_majority_class = []
+            n_minority_class = []
+            class_imbalance = []
+        elif isinstance(data, BaseRegressionDataset):
+            mean_label = []
+            std_label = []
+            max_value = []
+            min_value = []
+        
+        data_dfs = [dataset.df for dataset in split_datasets]
+        aggregated_data_df = pd.concat(data_dfs, axis=0, ignore_index=True)
+        data_dfs.append(aggregated_data_df)
+        target_column = split_datasets[0].target_column
+        for data_df in data_dfs:
```

Comment:
The following cases may be simplified if we add an abstract `summary` method (or similar) to the `AbstactDataset` that is implemented by all base datasets. This way we can keep the functionality of the `Bookkeeping` class to just exporting / saving to disk here, and if needed, custom datasets could override the summary. What are your thoughts?

Reply:
Yea, that might be a good idea. But shall we add it to the `AbstactDataset` or rather individually to all the task-specific abstract datasets like `BaseClassificationDataset`?

Reply:
We should add an abstract method to the abstract dataset and implement the defaults you have here in the base dataset classes. If specific features are needed then, a dataset could just override the method again.

Reply:
I am also in favor of this approach, and this covers the "modularity" concern I also express in my review